client/src/extension.ts

```diff
@@ -116,16 +118,82 @@ export function activate(context: ExtensionContext) {
         // nothing more to do
         return;
     }
+
+    // First check recursively all the files set in `project.kao`-like files, starting with `project.kao`.
+    workspace.findFiles(`**/project.kao`).then((uris) => {
+        uris.forEach((uri) => {
+            openProjectFile(uri);
+            // There should be only one `project.kao` file.
+            return;
+        });
+    });
+
     /*
-     * List of all the yseopml file extensions known by this extension as set in `client/package.json`.
+     * List of all the yseopml file extensions known by this extension as set in `client/package.json`
+     * but never explictly set in project.kao-like files.
      */
-    const yseopmlExtensions = ['kao', 'yclass', 'yobject', 'ycomplete', 'dcl', 'yml'];
-
+    const yseopmlExtensions = ['yclass', 'yobject', 'ycomplete'];
     for (const extension of yseopmlExtensions) {
         parseFilesWithExtension(extension);
     }
 }
 
+/**
+ * Try to open a file with URI `fileUri`.
+ * Then, if the file is a `project.kao`-like file (i.e. is a list files used for the project),
+ * read its content and apply this function recursivly for each line that are existing file's URI.
+ *
+ * @param fileUri An existing file URI.
+ */
+function openProjectFile(fileUri: Uri): void {
+    // Try to open the file. If it is opened, the server will parse it.
+    workspace.openTextDocument(fileUri).then(
+        // The document exists and was successfully opened and should be parsed already.
+        (doc) => {
+            console.debug(`Parsing ${fileUri}.`);
+            const lines = doc.getText().split('\n');
+            if (lines.length === 0 || !lines[0].trim().startsWith('_FILE_TYPE_')) {
+                // We are not in a `project.kao`-like file. Stop the going further.
+                return;
+            }
+            lines.forEach((line) => {
+                // line can be indented in the file.
+                line = line.trim();
+
+                /*
+                 * Ignore:
+                 * - any empty line
+                 * - lines that are just preprocessing stuff or Yseop Engine instruction.
+                 * - every file from .generated-yml/ directory
+                 */
+                if (
+                    line.length === 0 ||
+                    line.indexOf('@') >= 0 ||
+                    line.indexOf('_FILE_TYPE_') >= 0 ||
+                    line.indexOf('.generated-yml') >= 0
+                ) {
+                    return;
+                }
+                const maybeFilePath = path.join(path.dirname(doc.uri.fsPath), line);
+                if (!fs.existsSync(maybeFilePath)) {
+                    // File doesn't exist. Go to the next line.
+                    return;
```

Comment:
I’m pondering whether to warn the dev in such cases… But maybe this would be a job for another part of the program, with, like, red markers in the GUI on the considered line in the file, etc.

Reply:
Clearly not. We might encounter comments in the file or some other weird things.
Also, this feature is done without really opening the file, so displaying markers is impossible.
Plus, sometimes there are preprocessor instructions saying “If the file XXX exists […]”. It is not really our goal to do the job of the compiler.

Reply:
Huuum… If we check file existences for each comment line, it might end up being _more costly_ than the previous implem, don’t you think? Couldn’t we add dumb filtering conditions to weed out at least some comments? You did so for preprocessing instructions.

Reply:
I added the one-line comments as filtered case.